guides/authentication-methods/oauth/microsoft.mdx

```diff
@@ -0,0 +1,353 @@
+
+<Frame>
+    <img
+        src="/images/oauth-providers/microsoft/token-configuration.png"
+        alt="View app token configuration"
+        width={150}
+    />
+</Frame>
+
+2. Select `Add optional claim`.
+
+<Frame>
+    <img
+        src="/images/oauth-providers/microsoft/add-optional-claim.png"
+        alt="Add new optional token claim"
+        width={500}
+    />
+</Frame>
+
+3. Select `ID` as the token type.
+4. Tick the `upn` checkbox to select it as an additional optional claim.
+5. Click `Add`.
+
+<Frame>
+    <img
+        src="/images/oauth-providers/microsoft/configure-optional-claim.png"
+        alt="Configure optional claim"
+        width={500}
+    />
+</Frame>
+
+6. Tick the checkbox to set the required Microsoft Graph permissions.
+7. Click `Add`.
+
+<Frame>
+    <img
+        src="/images/oauth-providers/microsoft/set-graph-permissions.png"
+        alt="Set Microsoft Graph permissions for the upn claim"
+        width={500}
+    />
+</Frame>
+
+### Protection against the nOAuth vulnerability
+
+In 2023 Descope [discovered](https://www.descope.com/blog/post/noauth) a vulnerability resulting
+from misconfigurations of multi-tenant apps that could potentially lead to account takeovers when linking
+accounts with the provider (account linking is activated by default in Hanko). To mitigate this
+vulnerability Microsoft provides the following guidance:
+
+1. Configure the `authenticationBehaviors` setting of your application to disallow unverified email domains. This should
+be the default case for newly created applications, so no further steps should be required.
+
+<Info>
+    If your application was created before the publication of the vulnerability, and the application previously used
+    unverified emails, you might need to manually alter the `authenticationBehaviours` as described
+    [here](https://learn.microsoft.com/en-us/entra/identity-platform/migrate-off-email-claim-authorization#how-do-i-protect-my-application-immediately).
+</Info>
+
+2. Use the optional `xms_edov` token claim to check if an email domain owner has been verified. We recommend implementing
+this additional security measure in case the tokens do not contain a `upn` (e.g. if the claim has not been configured
+or the configuration has been reverted) and decisions have to be based on a users' `email` property. To do so, first open
+your applications `Manifest`.
+
+<Frame>
+    <img
+        src="/images/oauth-providers/microsoft/manifest.png"
+        alt="View app manifest"
+        width={150}
+    />
+</Frame>
+
+3. Find the `optionalClaims` key and set its value to  the following (make a backup of the manifest file, just in case):
+
+```json
+"optionalClaims": {
+		"idToken": [
+			{
+				"name": "email",
+				"source": null,
+				"essential": false,
+				"additionalProperties": []
+			},
+			{
+				"name": "xms_edov",
+				"source": null,
+				"essential": false,
+				"additionalProperties": []
+			}
```

Comment:
it seems that `xms_edov` is not longer supported. At least I got a warning sign in my `Token configuration` pane that the claim will not be included into the token because it is not supported